src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -95,18 +97,25 @@
   private final RepositoryRemoteExecutorFactoryDelegate repositoryRemoteExecutorFactoryDelegate =
       new RepositoryRemoteExecutorFactoryDelegate();
 
+  private final RemoteDownloaderFactory remoteDownloaderFactory = new RemoteDownloaderFactory();
```

Comment:
Not asking you to make any changes right now, but why is this part of the RemoteModule? Would it make sense to use the remote downloader without remote execution?

Reply:
`RemoteModule` handles all three remote capabilities: execution, caching, and downloading.

Remote downloading has a hard dependency on the remote cache, because that's where it puts downloaded content for consumption by Bazel.